pkg/pool-weighted/contracts/managed/ManagedPool.sol

```diff
@@ -244,14 +230,20 @@ contract ManagedPool is ManagedPoolSettings {
         uint256[] memory scalingFactors = _scalingFactors(tokens);
         _upscaleArray(balances, scalingFactors);
 
-        uint256 preJoinExitSupply = _beforeJoinExit(totalSupply());
+        // The AUM fee calculation is based on inflating the Pool's BPT supply by a target rate.
+        // We then must collect AUM fees whenever joining or exiting the pool to ensure that LPs only pay AUM fees
+        // for the period during which they are an LP within the pool: otherwise an LP could shift their share of the
+        // AUM fees onto the remaining LPs in the pool by exiting before they were paid.
+        uint256 preFeeCollectionSupply = totalSupply();
+        (uint256 protocolAUMFees, uint256 managerAUMFees) = _collectAumManagementFees(preFeeCollectionSupply);
+        uint256 postFeeCollectionSupply = preFeeCollectionSupply + protocolAUMFees + managerAUMFees;
```

Comment:
The safemath gods are not pleased by this

Reply:
The safemath gods are sated by a previous offering from when these fees were minted.